app/graphql/types/profile.rb

```diff
@@ -68,19 +69,23 @@ def total_host_count
       object.hosts.count
     end
 
-    def compliant(system_id:)
+    def compliant(args = {})
+      system_id = args[:system_id] || context[:parent_system_id]
```

Comment:
It seems like you should just go ahead and make this a private method with a cached (`||=`) instance variable.

Reply:
I can make it a private method, but I don't think caching is appropriate here because you may use the argument on certain methods, and rely on context for others.

Reply:
Gotcha, that makes sense